packages/components/src/modal/index.tsx

```diff
@@ -198,7 +198,13 @@ function UnforwardedModal(
 								) }
 							</div>
 						) }
-						{ children }
+						<div
+							className="components-modal__container"
+							tabIndex={ 0 }
+							role="region"
```

Comment:
This region is not going to work here I do not think.

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/dialog_role

Does not look like this would really be a valid approach. I wonder what would happen without the role here? Might be worth a try. If we introduce this ARIA here, we'll have to introduce it elsewhere. E.g. Using `role="banner"`, `role="main"`, `role="contentinfo"`. Having one region like this in my opinion would be too confusing.